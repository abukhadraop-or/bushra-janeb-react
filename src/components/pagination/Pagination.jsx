import _ from 'lodash';
import PropTypes from 'prop-types';
import React from 'react';
import {
  Container,
  PageButton,
  UnorderedListPagination,
} from './pagination.styles';

/**
 * Render the <Pagination> component.
 *
 * @param {Object} props              Props.
 * @param {number} props.currentPage  The current page that the user is in.
 * @param {number} props.itemCount    Number of articles.
 * @param {func}   props.onPageChange Function that handles the page change when clicking on the pagination.
 * @param {number} props.pageSize     Capacity of the page.
 *
 * @return {JSX.Element}
 */
export const Pagination = ({
  currentPage,
  itemCount,
  onPageChange,
  pageSize,
}) => {
  const pagesCount = Math.ceil(itemCount / pageSize);

  if (pagesCount === 1) return null;
  const pages = _.range(1, pagesCount + 1);

  return (
    <Container>
      <UnorderedListPagination>
        <li>
          {pages.map((page) => (
            <PageButton
              isActive={page === currentPage}
              key={page}
              onClick={() => {
                onPageChange(page);
              }}
            >
              {page}
            </PageButton>
          ))}
        </li>
      </UnorderedListPagination>
    </Container>
  );
};

Pagination.propTypes = {
  itemCount: PropTypes.number.isRequired,
  pageSize: PropTypes.number.isRequired,
  onPageChange: PropTypes.func.isRequired,
  currentPage: PropTypes.number.isRequired,
};

export default Pagination;
