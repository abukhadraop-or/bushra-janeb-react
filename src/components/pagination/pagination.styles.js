import styled from "styled-components";

export const Container = styled.div`
  margin: 0 auto;
  max-width: 62rem;
  padding: 0 1rem;
`;

export const UnorderedListPagination = styled.ul`
  border-radius: 0.25rem;
  display: flex;
  list-style-type: none;
  margin-bottom: 1rem;
  margin-top: 1rem;
  padding-left: 0;
`;

export const PageButton = styled.button`
  background-color: ${(props) => (props.isActive ? `#5cb85c` : `#fff`)};
  border: 0.063rem solid #ddd;
  color: ${(props) => (props.isActive ? `#ffffff` : `#5cb85c`)};
  cursor: pointer;
  margin-left: -1px;
  padding: 0.5rem 0.75rem;
  text-decoration: none;
  &:hover {
    background-color: #eceeef;
  }
  &:first-child {
    border-bottom-left-radius: 0.5rem;
    border-top-left-radius: 0.5rem;
  }

  &:last-child {
    border-bottom-right-radius: 0.5rem;
    border-top-right-radius: 0.5rem;
  }
`;
