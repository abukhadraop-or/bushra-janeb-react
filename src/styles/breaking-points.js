const breakpoints = {
  // 360px
  mobile: '20rem',

  // 480px
  mobileWide: '21.438rem',

  // 650px
  tablet: '54.625rem',

  // 768px
  tabletWide: '48rem',

  // 980px
  desktop: '60rem',

  // 1180px
  desktopWide: '73.75rem',

  // 10444px
  desktopMax: '65.25',

  // 1092x
  desktopExtraWide: '68.25rem',

  // 1180px.
  LargeScreen: '73.75rem',
};

export default breakpoints;
