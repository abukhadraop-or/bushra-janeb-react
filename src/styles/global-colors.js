const colors = {
  black: '#000',
  concrete: '#f3f3f3',
  darkBlue: '#00000066',
  fern: '#5cb85c',
  Gray90: '#E5E5E5',
  lightGrey: '#bbbbbb',
  midnightBlue: '#0000004d',
  transparent: '#ffffff00',
  white: '#ffffff',
  veryLightGrey: '#ddd',
  aliceBlue: '#eceeef',
  aluminium: '#818a91',
};

export default colors;
