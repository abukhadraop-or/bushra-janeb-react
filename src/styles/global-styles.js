import { createGlobalStyle } from 'styled-components';

const GlobalStyle = createGlobalStyle`
  body {
    margin: 0;
    padding: 0;
    font-family: Open-Sans, Helvetica, Sans-Serif;
  }

  *,
  *::before,
  *::after {
    box-sizing: border-box;
  }
`;

export default GlobalStyle;
