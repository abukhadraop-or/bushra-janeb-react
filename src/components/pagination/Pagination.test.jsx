import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Enzyme, { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { Pagination } from './Pagination';
import { PageButton } from './pagination.styles';

Enzyme.configure({ adapter: new Adapter() });

describe('Pagination', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(
      <Pagination
        currentPage={1}
        itemCount={41}
        onPageChange={jest.fn()}
        pageSize={10}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('calls onPageChange on click', () => {
    const wrapper = mount(
      <Pagination
        currentPage={1}
        itemCount={41}
        onPageChange={jest.fn()}
        pageSize={10}
      />
    );
    const button = wrapper.find(PageButton).at(3);
    button.simulate('click');
    expect(wrapper.props().onPageChange).toBeCalled();
  });
});
