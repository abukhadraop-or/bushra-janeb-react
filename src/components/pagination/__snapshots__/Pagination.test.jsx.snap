// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should match snapshot 1`] = `
<Fragment>
  <styled.div>
    <styled.ul>
      <li>
        <styled.button
          isActive={true}
          key="1"
          onClick={[Function]}
        >
          1
        </styled.button>
        <styled.button
          isActive={false}
          key="2"
          onClick={[Function]}
        >
          2
        </styled.button>
        <styled.button
          isActive={false}
          key="3"
          onClick={[Function]}
        >
          3
        </styled.button>
        <styled.button
          isActive={false}
          key="4"
          onClick={[Function]}
        >
          4
        </styled.button>
        <styled.button
          isActive={false}
          key="5"
          onClick={[Function]}
        >
          5
        </styled.button>
      </li>
    </styled.ul>
  </styled.div>
</Fragment>
`;
